Processor_CashFlows_SftpCommunication.php

```diff
@@ -0,0 +1,64 @@
+<?php
+class Processor_CashFlows_SftpCommunication extends Jobs_SftpCommunication
+{
+    public function __construct($JobType, $FilenamePattern, $LogJob, $Verbose)
+    {
+        parent::__construct(
+            $JobType,
+            $FilenamePattern,
+            $LogJob,
+            Jobs_Chargeback_Cashflows::CASHFLOWS_SFTP_FILE_LIVESPAN,
+            $Verbose
+        );
+        $this->ProcessorName = 'CashFlows';
+        $this->setDoneDir(Jobs_Chargeback_Cashflows::DIR_TYPE_DONE);
+        $this->setCacheDir(Jobs_Chargeback_Cashflows::DIR_TYPE_CACHE);
+    }
+
+    public function setCommand($cmdDirectory)
+    {
+        $command = $cmdDirectory . DIRECTORY_SEPARATOR . "cashflows.sftp.exp " .
+            Jobs_Chargeback_Cashflows::CASHFLOWS_SFTP_SERVER . ' ' .
+            Jobs_Chargeback_Cashflows::CASHFLOWS_SFTP_PORT . ' ' .
+            Jobs_Chargeback_Cashflows::CASHFLOWS_SFTP_SERVER_USERNAME . ' ' .
+            Jobs_Chargeback_Cashflows::CASHFLOWS_SFTP_SERVER_PASSWORD;
+            if ($this->Verbose) echo "SFTP command: " . $command . " \n ";
+        parent::setCommand($command);
+    }
+    protected function getJobDirectory($JobType, $DirectoryType)
+    {
+        switch ($JobType) {
```

Comment:
If this line isn't to be remove in a refactor then this function should either be a static function which accepts `$JobType` or it should make use of `$this->JobType` from state.